WordPress/Classes/ViewRelated/Post/PostSyncStateViewModel.swift

```diff
@@ -0,0 +1,72 @@
+import Foundation
+
+final class PostSyncStateViewModel {
+    enum State {
+        case idle
+        case syncing
+        case offlineChanges
+        case failed
+    }
+
+    private let post: Post
+    private let isInternetReachable: Bool
+
+    var state: State {
+        if post.remoteStatus == .pushing || PostCoordinator.shared.isDeleting(post) {
+            return .syncing
+        }
+        if post.isFailed {
+            return isInternetReachable ? .failed : .offlineChanges
```

Comment:
It works in this iteration, considering it's a UI/UX change.
In terms of refactoring, I would suggest to:

- Remove the upload  status from the database and add it to PostCoordinator
- Instead of `isFailed` and `isInternetReachable` show "Offline Mode" when the uploads fail with a `.noInternetConnection` error